src/stratigraphy/util/coordinate_extraction.py

```diff
@@ -0,0 +1,283 @@
+"""This module contains the CoordinateExtractor class."""
+
+import abc
+import logging
+from dataclasses import dataclass
+
+import fitz
+import regex
+
+logger = logging.getLogger(__name__)
+
+COORDINATE_ENTRY_REGEX = "(?:[12][\.\s']{0,2})?\d{3}[\.\s']{0,2}\d{3}\.?\d?"
+
+
+@dataclass
+class CoordinateEntry:
+    """Dataclass to represent a coordinate entry."""
+
+    first_entry: str | None = None
+    second_entry: str | None = None
```

Comment:
Why do we need these at all? If it's just for string formatting purposes, then we should rather use [format specifications](https://docs.python.org/3/library/string.html#format-specification-mini-language) (use `,` and then replace with `'`) or some string operations (slicing and `"'".join()` ) instead.

Reply:
We usually extract two strings from the text and I wondered where we should process the strings. I decided it could be prettier to do it inside the class, than in the extraction code. That's why I kept the first and second entry, but introduced the coordinate value. The formatting was a nice side effect. Also, it was neat to deal with both coordinate formats the same way.

Either way, maybe the code gets a bit simpler if we slice and join in the extraction part.